libs/datamodel/core/src/transform/ast_to_dml/db/attributes.rs

```diff
@@ -203,19 +204,23 @@ fn visit_scalar_field_attributes<'ast>(
 
          // @unique
          attributes.visit_optional_single("unique", ctx, |args, ctx| {
-             if let Some(pk) = &model_data.primary_key{
-
-            if pk.source_field == Some(field_id) {
-                ctx.push_error(args.new_attribute_validation_error(
-                    "Fields that are marked as id should not have an additional @unique.",
-                ))
-            } }
+             if let Some(source) = &ctx.db.datasource{
+                if source.active_provider == "sqlserver" {
+                    if let Some(pk) = &model_data.primary_key{
+                        if pk.source_field == Some(field_id) {
+                            ctx.push_error(args.new_attribute_validation_error(
+                         "SQLServer only allows either a unique or a primary key constraint on a column.", ))
+                        }
+                    }
+                }
+             }
```

Comment:
This is fine but I am wondering what the ideal validation code here would look like. I think the `if source.active_provider == "sqlserver"` shouldn't use a magic string, and the provider name constant isn't much better. I'm thinking it could be part of the datamodel connector trait? It's ok not to do it now though because that would leak internals of `db` module that probably are going to change.

Also something weird is going on with rustfmt, nothing gets formatted in these files, it looks like (same on my machine).